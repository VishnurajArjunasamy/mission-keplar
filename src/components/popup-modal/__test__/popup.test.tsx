import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import PopUp from "../popup";

const mockStore = configureStore([]);

describe("Popup component", () => {
  let store: ReturnType<typeof mockStore>;

  beforeEach(() => {
    store = mockStore({
      sideBar: {
        isDarkMode: false,
      },
    });
  });

  it("renders modal with notification text and button texts", () => {
    render(
      <Provider store={store}>
        <PopUp />
      </Provider>
    );
    expect(
      screen.getByText("Editing in Progress. Do you want to leave?")
    ).toBeInTheDocument();
    expect(screen.getByText("YES")).toBeInTheDocument();
    expect(screen.getByText("NO")).toBeInTheDocument();
    expect(screen.getByText("X")).toBeInTheDocument();
  });

  it("renders modal with dark styles", () => {
    store = mockStore({
      sideBar: {
        isDarkMode: true,
      },
    });
    render(
      <Provider store={store}>
        <PopUp />
      </Provider>
    );
    expect(screen.getByText("X").parentElement?.parentElement).toHaveClass(
      "dark"
    );
  });

  it("calls handle handleBlogSwitch when Yes clicked", () => {
    const handleBlogSwitch = jest.fn();
    render(
      <Provider store={store}>
        <PopUp handleBlogSwitch={handleBlogSwitch} />
      </Provider>
    );
    const yesBtn = screen.getByText("YES");
    fireEvent.click(yesBtn);
    expect(handleBlogSwitch).toHaveBeenCalled();
  });

  it("calls handle handleCancel when No clicked", () => {
    const handleCancel = jest.fn();
    render(
      <Provider store={store}>
        <PopUp handleCancel={handleCancel} />
      </Provider>
    );
    const noBtn = screen.getByText("NO");
    fireEvent.click(noBtn);
    expect(handleCancel).toHaveBeenCalled();
  });

  it("calls handle closeModal when X clicked", () => {
    const closeModal = jest.fn();
    render(
      <Provider store={store}>
        <PopUp closeModal={closeModal} />
      </Provider>
    );
    const closeBtn = screen.getByText("X");
    fireEvent.click(closeBtn);
    expect(closeModal).toHaveBeenCalled();
  });

  it("Matches Snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <PopUp />
      </Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
