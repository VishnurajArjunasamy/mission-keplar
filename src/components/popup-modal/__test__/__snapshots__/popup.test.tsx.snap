// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component Matches Snapshot 1`] = `
<DocumentFragment>
  <div
    class="popUp light"
  >
    <div
      class="content"
    >
      <div
        class="close"
      >
        X
      </div>
      <p>
        Editing in Progress. Do you want to leave?
      </p>
      <div
        class="buttons"
      >
        <div
          class="button"
        >
          <button
            class="button"
            type="button"
          >
            YES
          </button>
        </div>
        <div
          class="button"
        >
          <button
            class="button"
            type="button"
          >
            NO
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
